opensearch_single_kernel/managers/health.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Base class for the OpenSearch Health management."""
+import time
+
+from tenacity import retry, stop_after_attempt, wait_fixed
+
+from opensearch_single_kernel.common.constants import HealthColors, StartMode
+from opensearch_single_kernel.common.exceptions import (
+    OpenSearchHAError,
+    OpenSearchHttpError,
+)
+from opensearch_single_kernel.core.state import ClusterState
+from opensearch_single_kernel.managers.base import BaseManager
+from opensearch_single_kernel.utils.topology import ClusterTopology
+from opensearch_single_kernel.workload.base import BaseWorkload
+
+
+class HealthManager(BaseManager):
+    """Class for managing OpenSearch statuses."""
+
+    def __init__(self, state: ClusterState, workload: BaseWorkload):
+        super().__init__(state, workload)
+
+    def get(  # noqa: C901
+        self,
+        wait_for_green_first: bool = False,
+        use_localhost: bool = True,
+        local_app_only: bool = True,
+    ) -> str:
+        """Fetch the current cluster status."""
+        if not (deployment_desc := self.state.application.deployment_desc):
+            return HealthColors.UNKNOWN
+
+        # the health depends on data nodes, for large deployments: an ML cluster
+        # may not be concerned about reporting or relying on the health of the
+        # data nodes in other clusters. We should therefore get this info from
+        # the deployment descriptor which has an overview of all the cluster.
+        # compute health only in clusters where data nodes exist
+        compute_health = (
+            deployment_desc.start == StartMode.WITH_GENERATED_ROLES
+            or ClusterTopology.is_data_role_in_cluster_fleet_apps(self.state)
+            or not local_app_only
+        )
+        if not compute_health:
+            return HealthColors.IGNORE
+
+        host = self._charm.unit_ip if use_localhost else None
+        response = self.opensearch_client.health(host, wait_for_green_first, self.alt_hosts)
+        if wait_for_green_first and not response:
+            response = self.opensearch_client.health(host, False, self.alt_hosts)
+
+        if not response:
+            return HealthColors.UNKNOWN
+
+        self.logger.info(f"Health: {response}")
+        try:
+            status = response["status"].lower()
+        except AttributeError as e:
```

Comment:
```suggestion
        except (AttributeError, TypeError, KeyError) as e: 
```
if response is an int, error will be TypeError.  If we try to catch above errors, it will be safer.